File: template_rust/src/pkg/sftp/domain/service.rs
package domain

import (
	"context"
	"template/pkg/sftp/domain/repository"
	"template/pkg/sftp/domain/usecase"
)

type Service struct {
	sftp repository.SftpHandler
}

func NewService(sftp repository.SftpHandler) usecase.SftpHandler {
	return &Service{
		sftp: sftp,
	}
}

func (s Service) Upload(ctx context.Context) error {
	return nil
}

func (s Service) Download(ctx context.Context) error {
	return nil
}

File: template_go_gorm/src/pkg/db/domain/service.go
package domain

import (
	"context"
	"template/pkg/db/domain/entity"
	"template/pkg/db/domain/repository"
	"template/pkg/db/domain/usecase"
)

type Service struct {
	store repository.Store
}

func NewService(
	store repository.Store,
) usecase.Store {
	return &Service{
		store: store,
	}
}

func (s Service) Find(ctx context.Context) ([]*entity.Entity, error) {
	return nil, nil
}
